### DISCUSSION TITLE
Using Google fonts with app directory

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#44906] [Help] [answered]
If I do 

```
<link href='https://fonts.googleapis.com/css2?family=Roboto+Mono&display=swap' rel='stylesheet' />
```

in my root head.js, it causes an hydration error.

I would strongly prefer to link to Google fonts like this, without using @next/font/google because I'm using styled components. However even if I suck it up and do

```
import { Roboto } from '@next/font/google'

const mono = Roboto({
  weight: '400',
  subsets: ['latin']
})
```

and 

```
	<style jsx global>
{`
  :root {
    --font-monospaced: ${mono.style.fontFamily};
  }
`}
</style>
```

this just doesn't work. It renders this html...

```
<style>
  :root {
    --font-monospaced: &#x27;__Roboto_19bf8f&#x27;, &#x27;__Roboto_Fallback_19bf8f&#x27; 
  }
</style>
```

But that var doesnt do anything.

How are google font supposed to work in the app directory?  The example app imports fonts into its page.tsx but you can't do this if you want to insert a style tag....

This doesn't seem to be in the docs anywhere. Thanks.

**Comment by ghost** [comment]
Well, I found a hacky way to achieve what I want. Not very satisfactory by any means but it works.

In `layout.js`

```
import Script from 'next/script'

const linkFont = `
	(() => {	
		var l = document.createElement('link');
		l.href = 'https://fonts.googleapis.com/css2?family=Roboto+Mono&display=swap' 
		l.rel='stylesheet'
		document.head.appendChild(l)		
	})();
`

export default function RootLayout({ children }) {
    return(
       ...
          <Script id="font">{linkFont}</Script>
       ...
    )
}

...



```

**Comment by hanneslund** [comment, marked as answer]
You can define the font as a CSS variable in the root layout like this:
```tsx

import { Roboto_Mono } from '@next/font/google'

const robotoMono = Roboto_Mono({
  subsets: ['latin'],
  variable: '--font-monospaced',
})

export default function RootLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html className={robotoMono.variable}>
      <body>{children}</body>
    </html>
  )
}
```

The suggested answer means that the font won't be self-hosted and also start to download very late.

**Reply by ghost** [reply]
Aha! This is very interesting, thank you. Can confirm it works needles to say. No doubt it's a much better solution for 99% of cases than the kludge I posted above!

That said Im bound to say that while the font magic is cool and all that it would be nice to have a simple way to opt out of magic if you have a good reason to. Like legacy components which can't be changed and which expect named font to be present. It's great to have @next/font, but it's a shame that it stops you using any other option. So I reluctantly have to prefer the grotty hack for my - doubtless unusual - case.
